test/plek_test.rb

```diff
@@ -97,14 +97,81 @@ def test_should_prepend_data_from_the_environment
     end
   end
 
+  def test_unprefixable_hosts_are_not_prefixed
+    ClimateControl.modify PLEK_HOSTNAME_PREFIX: "draft-",
+                          PLEK_UNPREFIXABLE_HOSTS: "signon,feedback" do
+      p = Plek.new("test.govuk.digital")
+      assert_equal "https://draft-content-store.test.govuk.digital", p.find("content-store")
+      assert_equal "https://signon.test.govuk.digital", p.find("signon")
+      assert_equal "https://feedback.test.govuk.digital", p.find("feedback")
+    end
+  end
+
+  def test_http_used_for_http_domains
+    ClimateControl.modify PLEK_HTTP_DOMAINS: "apps.svc.cluster.local" do
+      p1 = Plek.new("test.govuk.digital")
+      p2 = Plek.new("apps.svc.cluster.local")
+      assert_equal "https://frontend.test.govuk.digital", p1.find("frontend")
+      assert_equal "http://frontend.apps.svc.cluster.local", p2.find("frontend")
+    end
+  end
+
+  def test_http_domains_match_with_trailing_dot
+    # PLEK_HTTP_DOMAINS should match a domain with or without its trailing dot.
+    ClimateControl.modify PLEK_HTTP_DOMAINS: "apps.svc.cluster.local" do
+      p = Plek.new("apps.svc.cluster.local.")
+      assert_equal "http://signon.apps.svc.cluster.local.", p.find("signon")
+    end
+  end
+
+  def test_dev_domain_is_http_if_no_http_domains_specified
+    p = Plek.new
+    assert_equal "http://signon.dev.gov.uk", p.find("signon")
+  end
+
   def test_scheme_relative_urls
     url = Plek.new("dev.gov.uk").find("service", scheme_relative: true)
     assert_equal "//service.dev.gov.uk", url
   end
 
   def test_should_return_external_domain
     ClimateControl.modify GOVUK_APP_DOMAIN_EXTERNAL: "baz.external" do
-      assert_equal "http://foo.baz.external", Plek.new.external_url_for("foo")
+      assert_equal "https://foo.baz.external", Plek.new.external_url_for("foo")
+    end
+  end
+
+  def test_accepts_empty_domain_suffix
+    p = Plek.new("")
+    assert_equal "https://content-store", p.find("content-store")
+  end
+
+  def test_accepts_empty_domain_suffix_via_environment
+    ClimateControl.modify GOVUK_APP_DOMAIN: "",
+                          GOVUK_APP_DOMAIN_EXTERNAL: "example.com" do
+      assert_equal "https://content-store", Plek.new.find("content-store")
+    end
+  end
+
+  def test_accepts_empty_domain_suffix_in_rails_prod_mode
+    ClimateControl.modify RAILS_ENV: "production",
+                          GOVUK_APP_DOMAIN: "",
+                          GOVUK_APP_DOMAIN_EXTERNAL: "example.com" do
+      assert_equal "https://content-store", Plek.new.find("content-store")
+    end
+  end
```

Comment:
I don't think we should add this test as it doesn't execute a different unit of code. We should probably have one that tests that a NoConfiguration exception occurs if GOVUK_APP_DOMAIN raises an error isn't set, but since you didn't add the code I don't think it's necessary you add one (it already has coverage in asset_root and website_root test files)